### PR TITLE
Use material design theme

### LINKED ISSUE #664: Epic: Use Material Design for GRIP UI
There is a JavaFX implementation of material design. We should do a POC to see if it is viable to change the UI to use it.

Additionally we may want to provide this as an alternative styling that you can swap between.

### PR DESCRIPTION
**Pull request by SamCarlberg**
[//]: # (Please ensure that the "Allow edits from maintainers" checkbox is checked. Thanks!)

Closes #664 

Makes dialogs appear centered on the main window, instead of centered on the primary screen (which was a problem on multi-monitor setups)

# Screenshots

**Empty window**
Empty

**Pipeline**
Pipeline

**Step detail**
HSL Threshold

**Palette**
Palette

**Analysis dialog**
Pipeline Analysis_005

**Deploy dialog**
For reference only, since deploy is going to be removed
Deploy_008

**Exception alert**
IllegalStateException_009

### CODE REVIEW

**Comment by JLLeitschuh** [comment]
Well, that's different. Not bad though. Is there a dark mode? Lol

File: ui/src/main/resources/edu/wpi/grip/ui/GRIP.css [hunk 1]
```diff
@@ -210,7 +322,7 @@ VBox.sockets {
 .property-sheet .property-pane {
     -fx-hgap: 4em;
     -fx-vgap: 1em;
-    -fx-padding: 0;
+    /*-fx-padding: 0;*/
```
Comment (JLLeitschuh) [inline ui/src/main/resources/edu/wpi/grip/ui/GRIP.css]: Do you want to remove this?

File: ui/src/main/resources/edu/wpi/grip/ui/pipeline/OutputSocket.fxml [hunk 2]
```diff
@@ -9,7 +9,7 @@
 <?import javafx.scene.layout.HBox?>
 <?import javafx.scene.layout.StackPane?>
 <HBox styleClass="socket"
-      fx:controller="edu.wpi.grip.ui.OperationController"
+      fx:controller="edu.wpi.grip.ui.pipeline.OutputSocketController"
```
Comment (JLLeitschuh) [inline ui/src/main/resources/edu/wpi/grip/ui/pipeline/OutputSocket.fxml]: ?

Comment (SamCarlberg) [reply ui/src/main/resources/edu/wpi/grip/ui/pipeline/OutputSocket.fxml]: `fx:controller` is necessary, and `OperationController` is for controlling the UI element displaying an entire operation. `OutputSocketController` is more appropriate.

This also makes the file in line with `InputSocket.fxml`, which already uses `InputSocketController` as its `fx:controller` attribute

File: ui/src/main/resources/edu/wpi/grip/ui/GRIP.css (hunk 1 above)
Comment (SamCarlberg) [reply ui/src/main/resources/edu/wpi/grip/ui/GRIP.css]: Yeah, nice catch

**Comment by SamCarlberg** [comment]
No dark mode yet. Maybe later

**Comment by JLLeitschuh** [comment]
@SamCarlberg can this be merged?

**Pull request merged**